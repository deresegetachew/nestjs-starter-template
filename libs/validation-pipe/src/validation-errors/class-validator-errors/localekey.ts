export enum localeKey {
    isNotEmpty = 'dto:is-not-empty',
    minLength = 'dto:min-length',
    isEmail = 'dto:is-email',
    isString = 'dto:is-string'
}