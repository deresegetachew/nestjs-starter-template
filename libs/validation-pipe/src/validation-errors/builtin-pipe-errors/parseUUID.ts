import { I18nError, LogLevel } from '@lib/common';
import { BadRequestException } from "@nestjs/common";
import { localeKey } from './localekey';



class ParseUUIDErrorI18n extends I18nError {
    constructor(message: string, variables: { [key: string]: string | number } = {}) {
        super(message, variables, LogLevel.Info);
    }
}

export class ParseUUIDError extends BadRequestException {
    public I18nError: ParseUUIDErrorI18n;
    constructor(field: string) {
        super();
        this.I18nError = new ParseUUIDErrorI18n(localeKey.parseUUID, { field });
    }
}

