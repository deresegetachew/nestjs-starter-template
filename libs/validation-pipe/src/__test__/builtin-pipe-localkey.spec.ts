import { localeKey } from '../validation-errors/builtin-pipe-errors/localekey';

describe('validation pipe error Locale keys', () => {
    it('matches enum definition snap shot', () => {
        const keys = JSON.stringify(localeKey);

        expect(keys).toMatchSnapshot()
    })
})