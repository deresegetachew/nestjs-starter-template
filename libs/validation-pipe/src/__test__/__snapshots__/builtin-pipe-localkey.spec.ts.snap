// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation pipe error Locale keys matches enum definition snap shot 1`] = `"{\\"parseUUID\\":\\"common:validation.parse-uuid\\",\\"parseint\\":\\"common:validation.parse-int\\",\\"parseBool\\":\\"common:validation.parse-bool\\",\\"parseArray\\":\\"common:validation.parse-array\\"}"`;
